### ISSUE TITLE
Not perceptually uniform?

### ISSUE THREAD

**Issue by endolith** [endolith/complex_colormap#16]
Instead of JCh, it looks like this should be constructed in CAM02-UCS space (expressed in cylindrical coordinates) which has a different lightness curve and chroma(/colorfulness?) curve.

https://github.com/matplotlib/matplotlib/issues/6033#issuecomment-412318447 and https://github.com/matplotlib/matplotlib/issues/6033#issuecomment-412367529

("the new uniform colour spaces based on CIECAM02 by Luo et al. (CAM02-UCS, CAM02-SCD and CAM02-LCD)", so CIECAM02 itself is not a uniform color space)

and even after fixing that, it might still not be technically correct to call it "perceptually uniform", but I think "perceptually uniform in lightness and hue" would be correct.

**Comment by endolith** [comment]
@njsmith Is it correct to say that `colorspacious.cspace_converter("sRGB1", "CAM02-UCS")` returns J′a′b′ triples, and that when I convert them to cylindrical coordinates they become J′M′h?

```
Mʹ = np.sqrt(aʹ**2 + bʹ**2)
h = np.degrees(np.arctan2(bʹ, aʹ))

aʹ = Mʹ * np.cos(np.radians(h))
bʹ = Mʹ * np.sin(np.radians(h))
```

[Kruse 2018](https://doi.org/10.1364/OL.43.002426):

![kruse 2018 quote](https://user-images.githubusercontent.com/58611/44240352-ae347680-a18b-11e8-81ea-3fbf4f8019a2.png)

[Masaoka 2013](https://doi.org/10.1364/JOSAA.30.000264):

![masaoka 2013](https://user-images.githubusercontent.com/58611/44240353-b1c7fd80-a18b-11e8-84b6-9d85a14d93f6.png)


So then my colormap will be linear J′ ("lightness") vs linear h ("hue"), while maximizing M′ ("colorfulness") (in various ways).

**Comment by njsmith** [comment]
> Is it correct to say that colorspacious.cspace_converter("sRGB1", "CAM02-UCS") returns J′a′b′ triples, and that when I convert them to cylindrical coordinates they become J′M′h?

Yes. The J′a′b′ and J′M′h terminology isn't quite standard, but I know what you mean and what you mean is correct :-).

**Comment by endolith** [comment]
@njsmith  Is there a more standard terminology?

**Comment by njsmith** [comment]
Not really :-/. The Luo et al paper uses more cumbersome terminology where J′ refers to an intermediate value instead of the final result, just to make everyone's life difficult. What colorspacious call J′ matches what Luo et al call "J′ / K_L". (Though for CAM02-UCS specifically, K_L is 1, so these are the same again...)

I would ignore all this and just use J′, but since you were asking about the terminology specifically I wanted to give you a heads up that others could possibly get confused.

**Comment by endolith** [comment]
@njsmith  Ok, thanks,
